ubuntu-advantage

```diff
@@ -1,166 +1,141 @@
+xLfU21k0/ISLG+HzA7tlJQs9PXvza+qoHVbmeoIvJkjfZImSRK5f2bRBsoBWQFL9
+bN8TzEpyBCwpYOVdgvAFDtyBXHZ9BoPFXv7PUEzgNO3TQVhuC89thsn92B83kw88
+QQZH2UsRgZY0NH73C9EtK6Vd/KucLhJsPOOKyaX8N1TAS0wmjFYLdXsPALZJP0Ot
+JRhnzQEVahaW9U2ZZF73TSx65m0XDXpL9fZ0f4RLVOjwiSi1xKwrK5Cz1J/0ia2K
+z65pICP9AL2QtFlXx4z48cZH2vkI30KPmrrUmzxt/nVrpPNzPqC6KIAtXHi28neM
+wdDshtz6WQFkJ5ANJ7oEfJRrpq2bW3C0cM9fVBLP2KXdAC6wX9Oi+zk0hgPwC5Li
+YA7HF3LOvtXJfmd356iz0/bpMDtDWdsbm3Wffu/+9Yif+y48cnZ+eD9Ui1U4ibXg
+kt7c5jRfciMzwe6Vz7PnJFzxgnL+X5qBq7uneIKLUJscPpS6JKKosbMOlnYivSBs
+DvnC/IFPQwZLVWhWoNjfJrYlpxv3U2ntvCCRsqo81ErayaROJ4WkndxkUafEcyJw
+8dnHqmEfQhF5z/3UFZQ/Z+Vx6zTGUWWrDz7K9GNGzBMAEQEAAYkCHwQYAQIACQUC
+WPppLwIbDAAKCRC0wq96Z8egJtznEACDzfGE50VMi9GwgOiWBxbXFLj4g7mHELfN
+jrjJbGprVDFw9OMNkhy1gMfhr5mMXAE9ybqkM9KOC/ckkgoGWoiIeMB16URgXUV0
+CG99QYX4WvhPlaQUGXRpPqBgr+kPXQzdqNyPI2ffjnoPNOoheHJyWVRRaM+1Cy7Y
+gqHJ61AFdugGvNHMhST7sueJxInP42ZeAEzhcOp3cktZhZUudwKsSRkTNHDrleHK
+mv5XAZzS+dZA5VD9Md07juohkYeI6xx8diIsIa8RnD7VgKtyzXcfyNxOH0VMd3cM
+9wjKJGTS9iyfJvaGS5+vGLzedVcQk94WnoOUN9L19XTD8eAoI511+eHbpM9d+8Uf
+Tvn7bfN7kyrWHGUDa88qsx2j3ooYy4t3rSRiXu+SEJU7mFkz5ppxHC1DM+hh6MqC
+cO8Uh3kuyj7FR3pXgTKBXoGDQNf3KB0jsoHdZa6z8uO4A6rMx2qpm/Ru3iLFlrAG
+EeAmQXg53gHIXVfxX153FhmiGy/2ZOw4VHQSkUKKtLnea7XwnHHIXS8ojPl+JB07
+MyIKWikGem/sobDpokcVDSv1jGRlxuVNe5Y88leqmvqVkQ1SNPdSBopEKY2x/6z1
+hjyMm0AWNnQShlt+ChQn9Uv/GgcrhSU3KiM/RkiAyGuW1BKSC7BW3IroRWilSHVQ
+2Df3PX3pNQ==
+=FHLy
+-----END PGP PUBLIC KEY BLOCK-----
+EOF
+}
+
+
+write_list_file() {
+    cat > "$REPO_LIST" <<EOF
+deb https://${1}@${REPO_URL}/ubuntu precise main
+# deb-src https://${1}@${REPO_URL}/ubuntu precise main
+EOF
+}
+
+
+enable_esm() {
+    import_gpg_key
+    write_list_file "$1"
+    echo 'Ubuntu ESM repository enabled.  Run "sudo apt-get update" to update lists.'
+}
+
+
+disable_esm() {
+    if [ -f "$REPO_LIST" ]; then
+        mv "$REPO_LIST" "${REPO_LIST}.save"
+        echo 'Ubuntu ESM repository disabled.  Run "sudo apt-get update" to update lists.'
+    else
+        echo 'Ubuntu ESM repository was not enabled.'
+    fi
+}
+
+is_esm_enabled() {
+    apt-cache policy | grep -Fq "$REPO_URL"
+}
+
+validate_token(){
+    echo "$1" | grep -q '^[^:]\+:[^:]\+$'
+}
+
+
+usage() {
+    cat >&2 <<EOF
+usage: ${SCRIPTNAME} [enable-esm|disable-esm]
+
+Enable or disable the Ubuntu Extended Security Maintenance archive.
+
+Parameters:
+ enable-esm <token>     enable the ESM repository
+ disable-esm            disable the ESM repository
+
+the <token> argument must be in the form "user:password"
+
+EOF
+}
+
+
+if [ "$USER" != "root" ]; then
+    echo 'This command must be run as root (try using sudo)' >&2
+    exit 2
+fi
+
+case "$1" in
+    enable-esm)
+        if ! validate_token "$2"; then
+            echo 'Invalid token, it must be in the form "user:password"' >&2
+            exit 3
+        fi
+        enable_esm "$2"
+        ;;
+
+    disable-esm)
+        disable_esm
+        ;;
+
+    is-esm-enabled)
+        is_esm_enabled
+        ;;
+
+    *)
+        usage
```

Comment:
hmm, would be nice to see usage without being root, perhaps call this before bailing out on that check?